crates/circuit/src/dag_circuit.rs

```diff
@@ -2631,6 +2632,96 @@ def _format(operand):
                         | [OperationRef::Instruction(_), OperationRef::StandardInstruction(_)] => {
                             Ok(inst1.py_op_eq(py, inst2)? && check_args() && check_conditions()?)
                         }
+                        [OperationRef::Unitary(op_a), OperationRef::Unitary(op_b)] => {
+                            match [&op_a.array, &op_b.array] {
+                                [ArrayType::NDArray(a), ArrayType::NDArray(b)] => {
+                                    Ok(relative_eq!(a, b, max_relative = 1e-5, epsilon = 1e-8))
```

Comment:
Less of a comment on this PR and more of a general question, but what is `max_relative` vs `epsilon`, exactly?

I spent some time looking at the docs for the `approx` crate, but I could quite grok it.

Reply:
Hah, what docs? It's very sparse from what I remember. 

The `relative_eq!` macro is basically checking something like: `abs(a - b) <= epsilon + max_relative * abs(b)`. The equivalent in python land would be `numpy.is_close()` where `epsilon` is the  `atol` argument and `max_relative` is `rtol`. You can look at the impl for the `RelativeEq` trait for f64 to get a feeling for how it works.